### ISSUE TITLE
Font "Berkeley Mono Trial" behaving as "auto"

### ISSUE THREAD

**Issue by spencerwilson** [kovidgoyal/kitty#7716]
**Describe the bug**
`font_family Berkeley Mono Trial` is unexpectedly acting identically as `font_family auto` 

**To Reproduce**
1. Get the font Berkeley Mono from [here](https://berkeleygraphics.com/typefaces/berkeley-mono/). I specifically have the [free trial font](https://berkeleygraphics.com/products/FX-050/), which comes with two files BerkeleyMonoTrial-Regular.ttf and BerkeleyMonoTrial-Regular.otf
2. Add either file (both ttf and otf produce the same end result) to the system via Font Book, or `cp` into ~/Library/Fonts/
3. Verify that `kitty +list-fonts` output includes

```
Berkeley Mono Trial
    Berkeley Mono Trial Regular
```

4. In ~/.configferen/kitty/kitty.conf set `font_family Berkeley Mono Trial`
5. Reload config
6. Observe no difference from `font_family auto` (which on my system is same as `font_family Menlo`)

**Screenshots**
n/a

**Environment details**
```
kitty 0.35.2 (ad328bfeaa) created by Kovid Goyal
Darwin doggo.local 21.6.0 Darwin Kernel Version 21.6.0: Mon Feb 19 20:24:34 PST 2024; root:xnu-8020.240.18.707.4~1/RELEASE_X86_64 x86_64
ProductName:	macOS ProductVersion:	12.7.4 BuildVersion:	21H1123
OpenGL: '4.1 INTEL-18.8.16' Detected version: 4.1
Frozen: True
Paths:
  kitty: /Applications/kitty.app/Contents/MacOS/kitty
  base dir: /Applications/kitty.app/Contents/Resources/kitty
  extensions dir: /Applications/kitty.app/Contents/Resources/Python/lib/kitty-extensions
  system shell: /bin/zsh
Loaded config files:
  /Users/ssw/.config/kitty/kitty.conf

Config options different from defaults:
cursor_shape_unfocused  3
enable_audio_bell       False
font_family             Menlo
font_size               14.0
touch_scroll_multiplier 5.0
Removed shortcuts:
	cmd+, →  edit_config_file
	cmd+enter →  new_window
	cmd+h →  hide_macos_app
	cmd+k →  clear_terminal to_cursor active
	cmd+m →  minimize_macos_window
	cmd+n →  new_os_window
	cmd+q →  quit
	cmd+t →  new_tab
	cmd+w →  close_tab
	ctrl+cmd+, →  load_config_file
	ctrl+cmd+f →  toggle_fullscreen
	opt+cmd+h →  hide_macos_other_apps
	opt+cmd+r →  clear_terminal reset active
	opt+cmd+s →  toggle_macos_secure_keyboard_entry
	shift+cmd+/ →  open_url https://sw.kovidgoyal.net/kitty/
	shift+cmd+[ →  previous_tab
	shift+cmd+] →  next_tab
	shift+cmd+d →  close_window
	shift+cmd+w →  close_os_window
Colors:
	active_border_color     #babbf1   
	active_tab_background   #ca9ee6   
	active_tab_foreground   #232634   
	background              #303446   
	bell_border_color       #e5c890   
	color0                  #51576d   
	color1                  #e78284   
	color10                 #a6d189   
	color11                 #e5c890   
	color12                 #8caaee   
	color13                 #f4b8e4   
	color14                 #81c8be   
	color15                 #a5adce   
	color2                  #a6d189   
	color3                  #e5c890   
	color4                  #8caaee   
	color5                  #f4b8e4   
	color6                  #81c8be   
	color7                  #b5bfe2   
	color8                  #626880   
	color9                  #e78284   
	cursor                  #f2d5cf   
	cursor_text_color       #303446   
	foreground              #c6d0f5   
	inactive_border_color   #737994   
	inactive_tab_background #292c3c   
	inactive_tab_foreground #c6d0f5   
	mark1_background        #babbf1   
	mark1_foreground        #303446   
	mark2_background        #ca9ee6   
	mark2_foreground        #303446   
	mark3_background        #85c1dc   
	mark3_foreground        #303446   
	selection_background    #f2d5cf   
	selection_foreground    #303446   
	tab_bar_background      #232634   
	url_color               #f2d5cf   

Important environment variables seen by the kitty process:
	PATH                                /Applications/kitty.app/Contents/MacOS:/usr/bin:/bin:/usr/sbin:/sbin
	LANG                                en_US.UTF-8
	SHELL                               /bin/zsh
	USER                                ssw
```

I'm not sure why it says I've removed any shortcuts. I have not. Here is the entire config:

```
font_family Berkeley Mono Trial
```

**Comment by kovidgoyal** [comment]
Font selection was completely re-written for https://github.com/kovidgoyal/kitty/issues/7570  so try the nightly build and if you still have an issue follow up there.

**Comment by spencerwilson** [comment]
Yes, this bug is fixed as of c852a56574 (current nightly). Thank you!

**Comment by daddydiaz2** [comment]
IoskeleyMono
